semtypes/src/main/java/io/ballerina/types/SemTypes.java

```diff
@@ -52,6 +55,11 @@ public static SemType booleanConst(boolean value) {
         return BooleanSubtype.booleanConst(value);
     }
 
+    public static SemType decimalConst(String value) {
+        BigDecimal d = new BigDecimal(value.substring(0, value.length() - 1));
```

Comment:
Does decimal always have a `123d` like pattern?

Reply:
I'll extend to handle other cases without suffix